src/algebra/ring/pi.lean

```diff
@@ -47,6 +47,11 @@ by refine_struct { zero := (0 : Π i, f i), one := 1, add := (+), mul := (*),
   nsmul := add_monoid.nsmul, npow := monoid.npow };
 tactic.pi_instance_derive_field
 
+instance non_unital_non_assoc_ring [∀ i, non_unital_non_assoc_ring $ f i] :
+non_unital_non_assoc_ring (Π i : I, f i) :=
+by refine_struct { add := (+) };
```

Comment:
You should list all data fields here. Last time I checked, the tactic succeeded but created bad data fields (I don't remember why they're bad).

Reply:
Thanks. Done.